### DISCUSSION TITLE
关卡选择逻辑优化

### DISCUSSION THREAD

**Discussion by Zirun-wang** [MaaAssistantArknights/MaaAssistantArknights#1313] [Show and tell]
(*皆基于3.9.6版，更高版本如有改动就当我没说)

1、逻辑优化
“上次作战”和“当前关卡”可以合并，原因如下
> 若不在关卡界面，则“当前关卡”无意义，实际上仅“上次作战”可用
> 大多数情况下，“上次作战”和“当前关卡”指同一关卡
> (*误选则会跳过清理智环节，用户体验有改进余地)

故可合二为一，首先检测是否存在“当前关卡”，存在即进入刷图流程，不存在则调用"上次作战"，确保理智清的干干净净

2、GUI优化
当前(3.9.6)关卡选择列表太长且不够直观，摸了一种方案供参考

![屏幕截图 2022-07-24 151311](https://user-images.githubusercontent.com/84625349/180636618-45e491f1-f82a-4448-b782-8fa1360656d4.png)

**Comment by YipKei** [comment]
补充一下，材料本可以再加个二级选单：
+ 效率优先
+ 主材优先

或者直接提供两到三个关卡选择（不过需要出现三个关卡的只有固源岩组）

**Reply by MistEO** [reply]
全关卡导航还没做呢（

**Comment by ABA2396** [comment]
可以在 `上次作战` 的task里先判断有没有 `开始行动` ，有的话直接结束（？）

**Reply by MistEO** [reply]
上次作战 的 task 被很多地方复用了，不太好改可能

**Reply by YipKei** [reply]
是时候重构屎山了（

**Reply by MistEO** [reply]
https://github.com/MaaAssistantArknights/MaaAssistantArknights/commit/bb07197823cbf4caafaabc83c088bcd7d0c62602 改完了
